spog/ui/src/components/vulnerability.rs

```diff
@@ -58,20 +64,20 @@ pub fn vulnerability_search(props: &VulnerabilitySearchProperties) -> Html {
     let search = {
         let service = service.clone();
         use_async_with_cloned_deps(
-            move |(state, offset, limit)| async move {
+            move |(state, page, per_page)| async move {
                 service
                     .search_vulnerabilities(
                         &state,
                         &SearchOptions {
-                            offset: Some(offset),
-                            limit: Some(limit),
+                            offset: Some((page - 1) * per_page),
```

Comment:
Just wondering if this should be (page - 1).max(0) to ensure we can't accidentally wrap around. Alternatively use checked_sub() https://docs.rs/num/latest/num/trait.CheckedSub.html.

Reply:
Thanks for the review. The `use_pagination_state` should make sure invalid pages are never propated https://github.com/trustification/trustification/pull/153/files#diff-c04258e6f855e5575e6a3be0b3de08e9e718d61cd489dadb130fd48400a65ca2R41

And I also apply your suggestion moving that small logic to a separate function here https://github.com/trustification/trustification/pull/153/files#diff-3cda560b7a8c01c60599ccf022c1af4535ccc822437dbb47db755ed6051da6afR44

Reply:
Looks good, nice work!